modules/standard/IO.chpl

```diff
@@ -2804,111 +3029,139 @@ record defaultSerializer {
         _arrayMax = _arrayDim;
     }
 
-    @chpldoc.nodoc
+    /*
+      End the current dimension.
+    */
     proc ref endDim() throws {
       _arrayDim -= 1;
       _first = true;
     }
 
-    @chpldoc.nodoc
+    /*
+      Serialize ``element``.
+
+      Adds a space if this is not the first element in the row.
+    */
     proc ref writeElement(const element: ?) throws {
       if !_first then writer._writeLiteral(" ");
       else _first = false;
 
       writer.write(element);
     }
 
-    @chpldoc.nodoc
+    /*
+      Ends serialization of the current array.
+    */
     proc endArray() throws { }
   }
 
-  // Array helpers
-  //
-  // BHARSH TODO: what should we do for sparse arrays? The current format is
-  // kind of weird. I'd personally lean towards writing them as a map.
-  @chpldoc.nodoc
-  proc startArray(writer: fileWriter, size: int) throws {
-    return new ArraySerializer(writer);
+  /*
+    Start serializing a map by writing a ``{``.
+
+    :arg writer: The ``fileWriter`` to be used when serializing.
+    :arg size: The number of entries in the map.
+
+    :returns: A new MapSerializer
+  */
+  proc startMap(writer: fileWriter, size: int) throws {
+    writer._writeLiteral("{");
+    return new MapSerializer(writer);
   }
 
+  /*
+    Returned by ``startMap`` to provide the API for serializing maps.
+
+    Maps are serialized as a comma-separated series of pairs between curly
+    braces. For example, the keys ``1``, ``2``, and ``3`` with values
+    corresponding to their squares would be serialized as:
+
+    ::
+
+      {1: 1, 2: 4, 3: 9}
+
+    Empty maps be serialized as ``{}``.
+  */
   record MapSerializer {
+    @chpldoc.nodoc
     var writer;
+    @chpldoc.nodoc
     var _first : bool = true;
 
-    @chpldoc.nodoc
+    /*
+      Serialize ``key``.
+
+      Adds a leading comma if this is not the first pair in the map.
+    */
     proc ref writeKey(const key: ?) throws {
       if !_first then writer._writeLiteral(", ");
       else _first = false;
 
       writer.write(key);
     }
 
-    @chpldoc.nodoc
+    /*
+      Serialize ``val``, preceded by a colon.
```

Comment:
Probably have ``:`` in there as well as most of the other `write...` methods have the characters in them.